src/MLJAbstractGPsGlue.jl

```diff
@@ -114,15 +120,21 @@ function predict(model::MLJAbstractGP, fit_result, X_new)
     return fit_result.posterior(x_new, fit_result.noise_variance + 1e-3)
 end
 
+function predict(model::MLJAbstractGP, fit_result, X_new)
+    return marginals(predict_joint(model, fit_result, X_new))
+end
+
 function predict_mean(model::MLJAbstractGP, fit_result, X_new)
-    return mean(predict(model, fit_result, X_new))
+    return map(mean, predict(model, fit_result, X_new))
 end
```

Comment:
No need for this, as the fallback for `predict_mean` implements precisely this, I believe. If you have a more performant alternative, then go ahead and implement it.